libs/wingsdk/src/target-tf-aws/table.ts

```diff
@@ -0,0 +1,110 @@
+import { DynamodbTable } from "@cdktf/provider-aws/lib/dynamodb-table";
+import { Construct } from "constructs";
+import { Function } from "./function";
+import * as cloud from "../cloud";
+import * as core from "../core";
+import { NameOptions, ResourceNames } from "../utils/resource-names";
+
+/**
+ * Counter (Table) names must be between 3 and 255 characters.
+ * You can use alphanumeric characters, dot (.), dash (-), and underscores (_).
+ */
+const NAME_OPTS: NameOptions = {
+  maxLen: 255,
+  disallowedRegex: /[^a-zA-Z0-9\_\.\-]+/g,
+};
+
+/**
+ * AWS implementation of `cloud.Table`.
+ *
+ * @inflight `@winglang/sdk.cloud.ITableClient`
+ */
+export class Table extends cloud.Table {
+  private readonly table: DynamodbTable;
+
+  constructor(scope: Construct, id: string, props: cloud.TableProps) {
+    super(scope, id, props);
+
+    if (this.columns[this.primaryKey] === undefined) {
```

Comment:
Yea if we use a Map (see comments in cloud.table.ts) we could change this to be someting like
```suggestion
    if (this.columns.has(this.primaryKey)) {
```